### PR TITLE
Variable refund notice

### PR DESCRIPTION
**Pull request by 0nko**
Addresses #2000.

Based on the available refund options (fees, shipping and taxes), the refund notice is changed accordingly.

| Single option | Two options | Three options |
|-|-|-|
| image | image | image |

**To test:**
1. Create orders with different combinations of items (fees, shipping and taxes)
2. Go to Order details and issue a refund
3. Verify the right combination of options is displayed in the notice (or the notice is hidden, if none are available)

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt [hunk 1]
```diff
@@ -173,6 +176,29 @@ class IssueRefundViewModel @AssistedInject constructor(
     }
 
     private fun initRefundByItemsState() {
+        fun getRefundNotice(): String? {
+            val refundOptions = mutableListOf<String>()
+            if (order.feesTotal > BigDecimal.ZERO) {
+                val fees = resourceProvider.getString(string.orderdetail_payment_fees).toLowerCase(Locale.ROOT)
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt]: ⚠️ I'm a bit worried "Locale.ROOT" might not work as expected in some languages. Have you considered using device's locale instead? Wdyt?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt [hunk 2]
```diff
@@ -173,6 +176,29 @@ class IssueRefundViewModel @AssistedInject constructor(
...
+            }
+            if (order.shippingTotal > BigDecimal.ZERO) {
+                val shipping = resourceProvider.getString(string.shipping).toLowerCase(Locale.ROOT)
+                refundOptions.add(shipping)
+            }
+            if (order.totalTax > BigDecimal.ZERO) {
+                val taxes = resourceProvider.getString(string.taxes).toLowerCase(Locale.ROOT)
+                refundOptions.add(taxes)
+            }
+            val and = resourceProvider.getString(string.and).toLowerCase(Locale.ROOT)
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt]: Nitpick: We could move "val and" into the `if (refundOptions.isNotEmpty())`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt [hunk 3]
```diff
@@ -6,6 +6,7 @@ import androidx.lifecycle.MutableLiveData
 import com.squareup.inject.assisted.Assisted
 import com.squareup.inject.assisted.AssistedInject
 import com.woocommerce.android.R
+import com.woocommerce.android.R.string
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt]: ⚠️ Importing `R.string` sometimes confuses lint when it tries to find usages of strings. We might want to consider removing this import. Wdyt?

**Review by malinajirka — CHANGES_REQUESTED** [review]
Thanks @0nko on working on this! 

I've tested the changes and it works as expected (with English locale). I've left a couple of comments, let me know what you think.

P.S. I realize this class is not unit tested, but we might want to consider creating a first few tests for this logic, wdyt?

**Comment by 0nko** [comment]
@malinajirka Thanks for the review! I've addressed your comments and the PR is ready for another round.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            networkStatus,
+            resourceProvider,
+            orderDetailRepository,
+            gatewayStore,
+            refundStore
+        )
+    }
+
+    @Test
+    fun `test order with no extra shipping options`() = coroutinesTestRule.testDispatcher.runBlockingTest {
```
Comment (malinajirka) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt]: 🔍 Nitpick: I think it's a good practice to explain what the test is testing and what is the expected result in the name of the test. Tests serve as a form of documentation and keeping the names self-explanatory helps all future devs to quickly understand what is the expected behavior.

We might for example name this something like `when order has zero taxes and no shipping and fees, then refund notice is not visible`.

Wdyt?

**Review by malinajirka — DISMISSED** [review]
Thanks for the changes @0nko! It all looks good to me now ;). I've left one minor comment, but feel free to merge it no matter if you decide to update the test names or not.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt (hunk 4 above)
Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt]: Thanks, good point 👍

Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt]: @malinajirka Fixed!

Comment (malinajirka) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModelTest.kt]: Love it! Thank you 🙇

**Review by malinajirka — APPROVED** [review]

**Pull request merged**